### PR TITLE
Add error tests

### PR DESCRIPTION
**Pull request by maririos**
Solves #81

### CODE REVIEW

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -276,7 +276,134 @@ public async Task UpdateIfETag()
...
+        [Test]
+        public async Task UpdateIfNoMatch()
+        {
+            var connectionString = Environment.GetEnvironmentVariable("AZ_CONFIG_CONNECTION");
+            Assert.NotNull(connectionString, "Set AZ_CONFIG_CONNECTION environment variable to the connection string");
+            var service = new ConfigurationClient(connectionString);
+
+            Response<ConfigurationSetting> responseSet = await service.SetAsync(s_testSetting, CancellationToken.None);
+            Assert.AreEqual(200, responseSet.Status);
+            Response<ConfigurationSetting> responseGet = await service.GetAsync(s_testSetting.Key, s_testSetting.Label, CancellationToken.None);
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Will this conflict with other tests that run in parallel? i.e. you are not using the GUID trick here.

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 2]
```diff
@@ -276,7 +276,134 @@ public async Task UpdateIfETag()
...
+            Assert.AreEqual(200, responseGet.Status);
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Is this needed? Don't we throw if status is not 200?

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 3]
```diff
@@ -276,7 +276,134 @@ public async Task UpdateIfETag()
...
+                var response = exception.Response;
+                Assert.AreEqual(412, response.Status);
+
+                responseSet.Dispose();
+                responseGet.Dispose();
+            }
+            finally
+            {
+                var responseDelete = await service.DeleteAsync(key: setting.Key, filter: setting.Label, CancellationToken.None);
+                if (responseDelete.Status != 200)
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: We throw if delete fails. No need to check the status.

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 4]
```diff
@@ -510,6 +637,40 @@ public async Task GetList()
...
+            var service = new ConfigurationClient(connectionString);
+
+            var testSettingNoLabel = s_testSetting.Clone();
+            testSettingNoLabel.Label = null;
+            Response<ConfigurationSetting> responseSet = await service.SetAsync(testSettingNoLabel, CancellationToken.None);
+            Assert.AreEqual(200, responseSet.Status);
+            
+            try
+            {
+                var response = await service.GetAsync(testSettingNoLabel.Key, "%00", CancellationToken.None);
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Could you add a TODO to this line of code. I really don't like how "%00" is a magic string representing a null label. I would like to think how to make it cleaner.

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs (hunk 2 above)
Comment (maririos) [reply Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: :| Removing

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs (hunk 3 above)
Comment (maririos) [reply Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Fixed

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs (hunk 4 above)
Comment (maririos) [reply Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Fixed

**Pull request merged**